src/Microsoft.Identity.Client/Azure/TokenProvider.cs

```diff
@@ -0,0 +1,43 @@
+using Microsoft.Identity.Client.Http;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.Azure
+{
+    /// <summary>
+    /// ITokenProvider describes the interface for fetching an access token
+    /// </summary>
+    public interface ITokenProvider
+    {
+        /// <summary>
+        /// GetTokenAsync will attempt to fetch a token for a given set of scopes
+        /// </summary>
+        /// <param name="scopes">Scopes requested to access a protected API</param>
+        /// <returns>An access token as a string</returns>
+       Task<IToken> GetTokenAsync(IEnumerable<string> scopes = null);
+    }
+
+    /// <summary>
+    /// IToken is an AAD access token with an expiration
+    /// </summary>
+    public interface IToken
```

Comment:
Would it make sense, from an API point of view to leverage IAuthenticationResult? or do we want to introduce another structure?

Reply:
The reason I introduced this is that the developer doesn't need to know about anything but an opaque token and an expiry. Anything more is superfluous.